### PR TITLE
Routing: UI

### PR DESCRIPTION
**Pull request by thiloSchlemmer**
Child components to implement:
- [x] CategoryBar
- [x] FeedbackBanner
- [x] RoutingInfo
- [x] Waypoints
- [x] RouteChart

Additional Components
- [X]  Chips (Export)

### CODE REVIEW

File: src/services/RoutingService.js [hunk 1]
```diff
@@ -80,6 +83,67 @@ import { bvvRouteStatsProvider } from './provider/routeStats.provider';
...
+export const CHART_ITEM_SURFACE_STYLE_UNKNOWN = {
+	id: 0,
+	color: 'transparent',
+	image: 'repeating-linear-gradient(45deg,gray 25%, transparent 25%,transparent 50%, gray 50%, gray 55%, transparent 55%, transparent)',
+	label: 'Unknown'
+};
+
+/**
+ * A function that maps and reduces {@link module:domain/routing~RouteDetailTypeAttribute}
+ * with OSM road type names to the name of defined {@link module:domain/routing~ChartItemStyle}
```
Comment (taulinger) [inline src/services/RoutingService.js:134]: Isn't it a method of the `RoutingService`?
`@param routeDetailTypeAttributes` is missing

File: src/services/provider/etaCalculation.provider.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * for specific vehicle classes (Hike, Bike,MTB.Racingbike)
+ * Based on formulas from DAV and DIN (DIN 33466) for hiking only
+ * but adapted for Bike, MTB and Racingbike
+ *
+ * walking duration estimate based on DAV-Normative:
+ *  - https://discuss.graphhopper.com/t/walking-duration-estimate-elevation-ignored/4621/4
+ *  - https://www.alpenverein.de/chameleon/public/908f5f80-1a20-3930-1692-41be014372d2/Formel-Gehzeitberechnung_19001.pdf
+ * @see https://de.wikipedia.org/wiki/Marschzeitberechnung
+ * @function
+ * @implements {module:services/RoutingService~etaCalculationProvider}
```
Comment (taulinger) [inline src/services/provider/etaCalculation.provider.js:26]: <s>@implements</s> `@type`

File: src/services/provider/chartItemStyles.provider.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
+/**
+ * @module services/provider/chartItemStyles_provider
+ */
+
+import { $injector } from '../../injection/index';
+import { CHART_ITEM_ROAD_STYLE_UNKNOWN, CHART_ITEM_SURFACE_STYLE_UNKNOWN } from '../RoutingService';
+
+/**
+ * @implements chartItemStylesProvider
```
Comment (taulinger) [inline src/services/provider/chartItemStyles.provider.js:9]: <s>@implements</s> `@type {module:services/RoutingService~chartItemStylesProvider}`

File: src/services/RoutingService.js [hunk 4]
```diff
@@ -96,6 +160,27 @@ import { bvvRouteStatsProvider } from './provider/routeStats.provider';
...
+/**
+ * A function that provides a ETA calculation for a defined categoryId (vehicle type).
+ * @function
+ * @name module:services/RoutingService~RoutingService#getETAFor
+ * @param {string} categoryId
+ * @param {Number} distance distance in meter
+ * @param {Number} elevationUp elevation upwards in meter
+ * @param {Number} elevationDown  elevation downwards in meter
+ * @returns {number|null} the ETA in milliseconds or null if no ETA for the given category is available.
+ */
```
Comment (taulinger) [inline src/services/RoutingService.js:172]: Let's move this interface method declaration to the other declarations above
Order should be

1.  RoutingService methods
2. RoutingService provider

File: test/service/RoutingService.test.js [hunk 5]
```diff
@@ -18,6 +31,9 @@ describe('BvvRoutingService', () => {
 			expect(instanceUnderTest._categoriesProvider).toEqual(bvvRoutingCategoriesProvider);
 			expect(instanceUnderTest._routeProvider).toEqual(bvvRouteProvider);
 			expect(instanceUnderTest._routeStatsProvider).toEqual(bvvRouteStatsProvider);
+			expect(instanceUnderTest._chartItemsStylesProvider).toEqual(bvvChartItemStylesProvider);
+			expect(instanceUnderTest._mapper).toEqual(bvvOsmRoadTypeMappingProvider);
+			expect(instanceUnderTest._etaCalculationProvider).toEqual(bvvEtaCalculationProvider);
```
Comment (taulinger) [inline test/service/RoutingService.test.js:36]: Custom provider tests are missing in the next test `initializes the service with custom provider`

File: test/service/RoutingService.test.js [hunk 6]
```diff
@@ -240,4 +322,26 @@ describe('BvvRoutingService', () => {
 			expect(result).toEqual(mockStats);
 		});
 	});
+
+	describe('getETAFor', () => {
+		/**
+		 * @implements {module:services/RoutingService~ETACalculator}
+		 */
+		// eslint-disable-next-line no-unused-vars
+		const etaCalculator = { getETAfor: (distance, elevationUp, elevationDown) => 42 };
```
Comment (taulinger) [inline test/service/RoutingService.test.js:331]: Obsolete

File: test/service/RoutingService.test.js [hunk 7]
```diff
@@ -240,4 +322,26 @@ describe('BvvRoutingService', () => {
...
+
+		it('provides a ETACalculator', () => {
```
Comment (taulinger) [inline test/service/RoutingService.test.js:333]: Outdated description

File: src/services/provider/osmRoadTypeMapper.provider.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+const bvvMapping = {
+	unknown: ['other', 'track_other', 'path_other', 'footway_other'],
+	path: ['path_grade3', 'path_grade4', 'path_grade5'],
+	track: ['track_grade2', 'track_grade3', 'track_grade4', 'track_grade5'],
+	footway: ['footway_grade2', 'footway_grade3', 'pedestrian', 'cycleway', 'path_grade1', 'path_grade2'],
+	street: ['track_grade1', 'residential', 'unclassified', 'tertiary', 'service'],
+	mainstreet: ['secondary', 'primary', 'motorway']
+};
+
+export const bvvOsmRoadTypeMappingProvider = (osmRoadClasses) => {
```
Comment (taulinger) [inline src/services/provider/osmRoadTypeMapper.provider.js:20]: jsdoc is missing for this provider

**Review by taulinger — COMMENTED** [review]
Well done!

What do you think about renaming

- RoutingChart -> RouteChart
- RoutingInfo -> RouteInfo
- RoutingWarnings -> RouteWarnings

, cause they are always related to the current route?

File: src/services/RoutingService.js (hunk 1 above)
Comment (thiloSchlemmer) [reply src/services/RoutingService.js:134]: Yes, i change that....but the provider should also reference this ChartItemStyle to provide (override) a i18n-depending label property

File: src/services/provider/chartItemStyles.provider.js (hunk 3 above)
Comment (taulinger) [reply src/services/provider/chartItemStyles.provider.js]: just `@type {module:services/RoutingService~chartItemStylesProvider}`

File: src/services/provider/etaCalculation.provider.js (hunk 2 above)
Comment (taulinger) [reply src/services/provider/etaCalculation.provider.js]: just `@type {module:services/RoutingService~etaCalculationProvider}`

File: test/modules/header/components/Header.test.js [hunk 9]
```diff
@@ -343,18 +346,28 @@ describe('Header', () => {
...
+	// 	it('shows a modal window with the showcase', async () => {
+	// 		const element = await setup();
+
+	// 		element.shadowRoot.querySelector('.header__modal-button').click();
+
+	// 		expect(store.getState().modal.data.title).toBe('Showcase');
+	// 		//we expect a lit-html TemplateResult as content
+	// 		expect(store.getState().modal.data.content.strings[0]).toBe('<ba-showcase></ba-showcase>');
+	// 	});
+	// });
```
Comment (taulinger) [inline test/modules/header/components/Header.test.js:369]: Let's delete the uncommented lines
https://github.com/ldbv-by/bav4-nomigration/blob/03f67afa7fed3c2ae7282f598fa6c00977f723c9/test/modules/header/components/Header.test.js#L359-L369

File: test/modules/routing/components/RoutingInfo.test.js
Comment (taulinger) [inline test/modules/routing/components/RoutingInfo.test.js:1]: Must be named `RouteInfo.test.js`

**Review by taulinger — APPROVED** [review]
Well done!

**Pull request merged**